pkg/state/state.go

```diff
@@ -372,10 +372,10 @@ var DefaultFetchOutputDirTemplate = path.Join(
 )
 
 func (st *HelmState) ApplyOverrides(spec *ReleaseSpec) {
-	if st.OverrideKubeContext != "" {
+	if st.OverrideKubeContext != "" && spec.KubeContext == "" {
 		spec.KubeContext = st.OverrideKubeContext
 	}
-	if st.OverrideNamespace != "" {
+	if st.OverrideNamespace != "" && spec.Namespace == "" {
```

Comment:
We might need to revert this change as per https://github.com/helmfile/helmfile/pull/499/files#r1020827982